docs/specs/add-kubernetes-support.md

```diff
@@ -0,0 +1,209 @@
+# Move to Upstream Kubernetes
+
+## Introduction
+The ansible-service-broker and all of the tooling around it
+(apb-examples, catasb, and ansible-playbook-bundles) specifically target
+OpenShift as the underlying COE (container orchestration engine) when it
+should be cluster agnostic.
+
+## Problem Description
+Issues have been coming up during testing because the broker does not target
+the latest code for either the service-catalog or the underlying cluster.
+Kubernetes is the upstream community for OpenShift and the broker should be
+consumuing the latest upstream code for development purposes.
+
+This change will improve the Broker development process by having the Broker
+constantly being developed and tested against the latest upstream code.
+
+The Broker's functionality won't change, but all the tooling around it will.
+
+## Split Out All Cluster logic
+OpenShift is Kubernetes plus a few features, API name changes, and tools.
+In the core ansible-service-broker code, ```pkg/cluster/kubernetes.go```
+should be able to account for about 80% of the cluster opertations required
+by the broker. The remaining ~20% will be held in ```pkg/cluster/openshift.go```.
+
+### Code
+Start by creating a new pkg for cluster objects, ```pkg/cluster/...```. From
+there, the cluster pkg will hold ```pkg/cluster/cluster.go```,
+```pkg/cluster/kubernetes.go```, and ```pkg/cluster/openshift.go```.
+
+The file ```pkg/cluster/cluster.go```  will provide an object that will abstract
+cluster logic.
+
+Using an abstration is an advantage because the broker code will only have a
+single path. It will call the cluster object asking for an operation. For
+example, to create a new Pod:
+```diff
+- k8scli.CoreV1().Pods(ns).Create(pod)
++ Cluster.Create(pod, ns)
+```
+
+The Cluster object will recieve the request and call the correct API.
+```diff
++ func Create(c interface{}, ns string) {
```

Comment:
Is there some method they both share that we could make an interface for? That would allow the compiler to make some guarantees.